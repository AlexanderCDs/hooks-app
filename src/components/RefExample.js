/**
 * @author: alexandercd
 * @description: 
 * @date: 03/octubre/2021
 * **/
import React, {useState} from 'react';
import { MultipleCustomHooks } from './MultipleCustomHooks'; 

const RefExample = () => { 
    const [show, setShow] = useState(false);
    return (
        <div className="animate__animated animate__backInDown">
            <h1>Ref Example</h1> 
            <hr></hr>
            {show && (
                <MultipleCustomHooks />
            )}
            
            <button 
                className="btn btn-primary mt-2"
                onClick={ () => {
                    setShow(!show)
                }}
            >
                Show/Hide
            </button>
        </div>
    )
}

export default RefExample
