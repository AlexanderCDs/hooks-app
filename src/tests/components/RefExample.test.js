/**
 * @author: alexandercd
 * @description: 
 * @date: 18/octubre/2021
**/
import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import RefExample from '../../components/RefExample';

describe('Test <RefExample />', () => {
   
    const wrapper = shallow( <RefExample />);

    test('should show success', () => { 
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('MultipleCustomHooks').exists() ).toBe( false );
    });

    test('should show component <MultipleCustomHooks />', () => {

        wrapper.find('button').simulate('click');
        expect( wrapper.find('MultipleCustomHooks').exists() ).toBe( true );
    });
});
