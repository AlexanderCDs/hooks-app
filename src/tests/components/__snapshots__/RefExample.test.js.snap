// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <RefExample /> should show success 1`] = `
<div
  className="animate__animated animate__backInDown"
>
  <h1>
    Ref Example
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-2"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
